cf-agent/simulate_mode.c

```diff
@@ -87,10 +85,47 @@ static void ManifestStatInfo(const struct stat *st)
     };
     printf("Size: %ju\n", (uintmax_t) st->st_size);
     printf("Access: (%04o/%s)  ", st->st_mode & CHMOD_MODE_BITS, mode_str);
-    printf("Uid: (%ju/%s)   ",
+
+    errno = 0;
+    struct passwd *owner = getpwuid(st->st_uid);
+    if (owner != NULL)
+    {
+        printf("Uid: (%ju/%s)   ",
            (uintmax_t)st->st_uid, owner->pw_name);
-    printf("Gid: (%ju/%s)\n",
-           (uintmax_t)st->st_gid, group->gr_name);
+    }
+    else
+    {
+        if (errno == 0)
+        {
+            Log(LOG_LEVEL_ERR, "Failed to get file owner name: non-existent user id '%ju'", 
+                (uintmax_t)st->st_uid);
```

Comment:
Here and in other places, our style guide says there should be a space after the type cast.

https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#formatting--whitespace